cpu/cc3200/periph/spi.c

```diff
@@ -0,0 +1,264 @@
+ * @param clk SPI bit rate
+ */
+void _spi_config(spi_t bus, spi_mode_t mode, spi_clk_t clk)
+{
+	volatile cc3200_spi_t *dev = spi(bus);
+	/* current value of the ctrl register is used as a starting point */
+	cc3200_reg_t tmp = dev->module_ctrl;
+
+	/* compute divider value */
+	cc3200_reg_t divider = (XTAL_CLK / clk) - 1;
+
+	/* set master mode with hardware chip select */
+	tmp &= ~(MCSPI_MODULCTRL_MS | MCSPI_MODULCTRL_SINGLE);
+
+	/* set SPI config */
+	/* TIs code is also using OR with Master/Slave mode */
+	/* since riot only supports master mode we can skip that */
+	tmp |= (spi_config[bus].config >> 24) & 0xFF;
+
+	/* write config */
+	dev->module_ctrl = tmp;
+
+	/* reset tmp and set IS, DPE0, DPE1 for master mode */
+	tmp = 0x1 << 16;
+
+	/* mask config and set clock divider granularity to 1 cycle */
+	tmp = (tmp &
+	       ~(MCSPI_CH0CONF_WL_M | MCSPI_CH0CONF_EPOL | MCSPI_CH0CONF_POL |
+		 MCSPI_CH0CONF_PHA | MCSPI_CH0CONF_TURBO)) |
+	      MCSPI_CH0CONF_CLKG;
+
+	/* The least significant four bits of the divider is used fo configure */
+	/* CLKD in MCSPI_CHCONF next eight least significant bits are used to */
+	/* configure the EXTCLK in MCSPI_CHCTRL */
+	tmp |= ((divider & 0x0000000F) << 2);
+	dev->ch0_ctrl = ((divider & 0x00000FF0) << 4);
+
+	/* set protocol, CS polarity, word length and turbo mode */
+	dev->ch0_conf = ((tmp | mode) | (spi_config[bus].config & 0x0008FFFF));
+}
+
+void spi_init(spi_t bus)
+{
+	/* assert(bus >= SPI_NUMOF); */
+	mutex_init(&locks[bus]);
+	/* trigger pin initialization */
+	spi_init_pins(bus);
+
+	/* enable clock */
+	switch (bus) {
+	case EXTERNAL_SPI:
+		ARCM->MCSPI_A1.clk_gating |= PRCM_RUN_MODE_CLK;
+		break;
+	case CC3100_SPI:
+		ARCM->MCSPI_A2.clk_gating |=
+			PRCM_RUN_MODE_CLK | PRCM_SLP_MODE_CLK;
+		break;
+	}
+
+	/* reset spi for the changes to take effect */
+	spi_reset(bus);
+}
+
+int spi_acquire(spi_t bus, spi_cs_t cs, spi_mode_t mode, spi_clk_t clk)
+{
+	if (bus >= SPI_UNDEF) {
+		return -1;
+	}
+
+	/* lock bus */
+	mutex_lock(&locks[bus]);
+
+	/* enable cs pin */
+	if (cs == GPIO_UNDEF) {
+		spi(bus)->ch0_conf &= ~MCSPI_CH0CONF_FORCE;
+	} else {
+		spi(bus)->ch0_conf |= MCSPI_CH0CONF_FORCE;
+	}
+	/* configure */
+	_spi_config(bus, mode, clk);
+
+	/* enable spi */
+	spi(bus)->ch0_ctrl |= MCSPI_CH0CTRL_EN;
+	return SPI_OK;
+}
+
+void spi_release(spi_t bus)
+{
+	if (bus >= SPI_NUMOF) {
+		return;
+	}
+	mutex_unlock(&locks[bus]);
+
+	/* disable spi */
+	spi(bus)->ch0_ctrl &= ~MCSPI_CH0CTRL_EN;
+}
+
+uint8_t spi_transfer_byte(spi_t bus, spi_cs_t cs, bool cont, uint8_t out)
+{
+	spi_transfer_bytes(bus, cs, cont, &out, 0, 1);
+	return SPI_OK;
+}
+
+void spi_transfer_bytes(spi_t bus, spi_cs_t cs, bool cont, const void *out,
+			void *in, size_t len)
+{
+	/* TODO: handle cs and cont for */
+	(void)cont;
+
+	/* disable or enable cs */
+	if (cs == GPIO_UNDEF) {
+		spi(bus)->ch0_conf &= ~MCSPI_CH0CONF_FORCE;
+	} else {
+		spi(bus)->ch0_conf |= MCSPI_CH0CONF_FORCE;
```

Comment:
since `MCSPI_CH0CONF_FORCE` has nothing to do with chip select I enable and disable it before every spi transfer independant of cs . I've added the missing cs code so that should be fine now.
Thank you